sweep/sweeper.go

```diff
@@ -30,13 +73,585 @@ type UtxoSweeperConfig struct {
 	// Signer is used by the sweeper to generate valid witnesses at the
 	// time the incubated outputs need to be spent.
 	Signer lnwallet.Signer
+
+	// PublishTransaction facilitates the process of broadcasting a signed
+	// transaction to the appropriate network.
+	PublishTransaction func(*wire.MsgTx) error
+
+	// Ticker is an external dependency that drives the batch window logic.
+	Ticker ticker.Ticker
+
+	// BatchWindowSec indicates the size of the batching time window in
+	// seconds.
+	BatchWindowSec int
+
+	// SweepTxConfTarget assigns a confirmation target for sweep txes on
+	// which the fee calculation will be based.
+	SweepTxConfTarget uint32
+
+	// Notifier is an instance of a chain notifier we'll use to watch for
+	// certain on-chain events.
+	Notifier chainntnfs.ChainNotifier
+
+	// ChainIO allows us to query the state of the current main chain.
+	ChainIO lnwallet.BlockChainIO
+
+	// HasSpendingTx allows the sweeper to check whether there is already a
+	// unconfirmed or confirmed wallet transaction for this output. It
+	// prevents unnecessary republication and address inflation.
+	HasSpendingTx func(op wire.OutPoint) (*wire.MsgTx, error)
+
+	// MaxInputsPerTx specifies the  maximum number of inputs allowed in a
+	// single sweep tx. If more need to be swept, multiple txes are created
+	// and published.
+	MaxInputsPerTx int
+}
+
+// Result is the struct that is pushed through the result channel. Callers
+// can use this to be informed of the final sweep result. In case of a remote
+// spend, ErrRemoteSpend is return in err.
+type Result struct {
+	Err error
+	Tx  *wire.MsgTx
 }
 
-// New returns a new UtxoSweeper instance.
+type sweeperInput struct {
```

Comment:
even though these types are not exported, it gets a bit hard to follow with no comments.